### ISSUE TITLE
Empty results

### ISSUE THREAD

**Issue by bootstraponline** [TestArmada/flank#3]
The local `results` folder is empty. There's also no link to the corresponding Firebase test run.

```
$ java -jar flank.jar app.apk test.apk com.example.ui.DeviceTest

No Local 'flank.tests' found. It's used to create shards with configurable durations.

Creating bucket: gs://apk-bucket-2017-05-25-12-35

Uploading: app.apk to bucket: gs://apk-bucket-2017-05-25-12-35

Uploading: test.apk to bucket: gs://apk-bucket-2017-05-25-12-35

2 shards will be executed on: Nexus6P

Executing shard 0:  com.example.ui.DeviceTest#displaysPageObjects

Executing shard 1:  com.example.ui.DeviceTest#checkAnimationsOff


Fetching results to: ~/code/tmp/results

Combined test execution time: 0 seconds

End time: 2017-05-25-12-35
```

**Comment by bootstraponline** [comment]
<img width="269" alt="apk_bucket_empty" src="https://cloud.githubusercontent.com/assets/1173057/26460396/8233de72-4147-11e7-96fd-1b65e32dc57d.png">

I see the folder has been created but it's empty.

**Comment by bootstraponline** [comment]
I see the apks get uploaded and then deleted. The empty folder remains on cloud storage.

**Comment by renas** [comment]
Thanks for reporting this. What happens if you set debug-prints:true in the
properties file?

**Comment by bootstraponline** [comment]
here's the log: 
- https://gist.github.com/bootstraponline/962cbf17bce6f515b919e98d03d59b27

looks like there are a lot of issues

**Comment by bootstraponline** [comment]
>  public static final String ROOT_FLANK_BUCKET = "gs://flank/";
>
> This bucket name is already in use. Bucket names must be globally unique. Try another name.

I think this is one of the issues. ^

**Comment by bootstraponline** [comment]
The `apk-bucket-*` convention will have the same issue as well. In the Firebase SDK they generate guaranteed unique bucket names. For example: `2017-05-25_16:51:20.616122_pHkM/`

**Comment by renas** [comment]
We thought that was the issue as well but we are not able to reproduce the
issue even when we are using different accounts. Are you able to run tests
through the Firebase console directly?

**Comment by bootstraponline** [comment]
Yep. We are running on firebase directly with no issue.

**Comment by renas** [comment]
We are able to reproduce your issue. This happens when the account and
project are not in sync. The project ID has to belong to the correct
Firebase account. If you do a:

gcloud config list

What do you see?

Please use:

gcloud auth login

To set gcloud to the correct google account.
Then please use:

gcloud config set project <idOfYourProject> //Make sure its the project ID
and not just the name

And then try again. Please let me know how it goes.

Best
Renas

**Comment by bootstraponline** [comment]
Thanks. I'll try that tomorrow.

**Comment by bootstraponline** [comment]
```
$ gcloud config list
[compute]
region = us-east1
zone = us-east1-b
[core]
account = jenkins@delta-essence-114723.iam.gserviceaccount.com
disable_usage_reporting = True
project = delta-essence-114723

Your active configuration is: [default]
```

**Comment by bootstraponline** [comment]
> ERROR: (gcloud) Invalid choice: 'firebase'.

I updated my gcloud SDK and now the tests appear to be running correctly. The naming still seems like an issue though. For example `gs://apk-bucket-2017-05-26-10-05/` will conflict with other runs that occur during that same minute.

**Comment by renas** [comment]
Thanks for letting me know. We'll update the date stamp so it's unique.

**Comment by bootstraponline** [comment]
I am still getting errors. Here is the full log from my most recent run:

- https://gist.github.com/bootstraponline/86f0aba2ac1f684b9e4aeb87b68e154e

```
Creating gs://flank/...
ServiceException: 409 Bucket flank already exists.
Copying file://flank.tests [Content-Type=application/octet-stream]...
/ [0 files][    0.0 B/  121.0 B]
AccessDeniedException: 403 Caller does not have storage.objects.create access to bucket flank.
```

The tests run now though and the results folder contains *.xml files.

**Comment by renas** [comment]
Thanks will have a look.

**Comment by renas** [comment]
The ServiceException has been fixed in 1.2.0.
